src/SimpleWebRequest.ts

```diff
@@ -213,8 +215,35 @@ export abstract class SimpleWebRequestBase<TOptions extends WebRequestOptions =
     protected static checkQueueProcessing() {
         while (requestQueue.length > 0 && executingList.length < SimpleWebRequestOptions.MaxSimultaneousRequests) {
             const req = requestQueue.shift()!!!;
-            executingList.push(req);
-            req._fire();
+            const blockPromise = (req._blockRequestUntil&& req._blockRequestUntil()) || SyncTasks.Resolved();
+            blockPromise.then(() => {
+                if (executingList.length < SimpleWebRequestOptions.MaxSimultaneousRequests) {
+                    executingList.push(req);
+                    req._fire();
+                } else {
+                    req._enqueue();
+                }
+            }, err => {
+                // fail the request if the block promise is rejected
+                req._respond('Error in _blockRequestUntil: ' + err.toString());
+            });
+        }
+    }
+
+    protected _removeFromQueue(): void {
+        // Pull it out of whichever queue it's sitting in
+        if (this._xhr) {
+            _.pull(executingList, this);
+        } else {
+            _.pull(requestQueue, this);
+        }
+    }
+
+    protected _assertAndClean(expression: any, message: string): void {
```

Comment:
`expression: boolean` instead of `any`. The caller can force a boolean as needed.

Reply:
assert.ok takes any, so I kept it consistent